Makefile

```diff
@@ -355,9 +355,12 @@ error-code:
 # A special target for building TiKV docker image.
 .PHONY: docker
 docker:
-	bash ./scripts/gen-dockerfile.sh | docker build \
+	bash ./scripts/gen-dockerfile.sh > Dockerfile
```

Comment:
Dirty file.

Reply:
it's desired, we need to generate dockerfile and tar file firstly, then executing docker build to avoid can not found tar file problem and this file will be removed by each CI build

https://stackoverflow.com/a/47978479

or do we have any better way make bash script ` proc1 | proc2`, don't execute proc2 until proc1 process full result?

Reply:
But it will not be removed if I execute it locally.

Reply:
I don't think we need to dynamically generate a Dockerfile image. If you like I can send a PR to make it static (some scripting before and after may be needed).

Reply:
@gregwebs surely~ it's glad to see you can help refine this Dockerfile generation :smile: 

I'm just a new learner to docker & k8s, but found tikv's `/run-build-image`(https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/build_tikv_image_comment/activity) can not works for a long months during try to deploy newest version tidb to k8s, so this PR just a patch to old script, maybe there are better way to build image than this outdated script(because our nightly image build everyday didn't meet "max depth exceeded", it must be built by other way :D )

Reply:
here is a PR with a static Dockerfile: https://github.com/tikv/tikv/pull/8830